### PR TITLE
input/keyboard: Add way to access currently pressed keys

### PR DESCRIPTION
**Pull request by Drakulix**
This is e.g. useful to simulate `KeyboardTarget::enter` events, when internally multiplexing a `KeyboardTarget` implementation.

### CODE REVIEW

File: src/input/keyboard/mod.rs [hunk 1]
```diff
@@ -556,6 +556,24 @@ impl<D: SeatHandler + 'static> KeyboardHandle<D> {
         );
     }
 
+    /// Iterate over the keysyms of the currently pressed keys.
+    pub fn with_pressed_keysyms<F>(&self, mut f: F)
+    where
+        F: FnMut(KeysymHandle<'_>),
```
Comment (i509VCB) [inline src/input/keyboard/mod.rs:562]: Would it be more useful to have the closure provided an iterator rather than running multiple times?

Comment (i509VCB) [inline src/input/keyboard/mod.rs:562]: Should this have a return value?

Comment (Drakulix) [reply src/input/keyboard/mod.rs:562]: Not sure, I just followed the other apis. Due to borrowing state behind a Mutex for the KeysymHandle, we sadly can't simply return an Iterator...

I suppose this is a more general question about how we want to approach the few `with_` functions, that we have?

Comment (Drakulix) [reply src/input/keyboard/mod.rs:562]: Maybe for the case of running the closure once with an iterator, but the semantics of merging results for running this multiple times, seem to be unnecessary complex.

Comment (Drakulix) [reply src/input/keyboard/mod.rs:562]: ping @i509VCB

File: src/input/keyboard/mod.rs [hunk 2]
```diff
@@ -580,6 +580,24 @@ impl<D: SeatHandler + 'static> KeyboardHandle<D> {
         );
     }
 
+    /// Iterate over the keysyms of the currently pressed keys.
+    pub fn with_pressed_keysyms<F>(&self, mut f: F)
```
Comment (bwidawsk) [inline src/input/keyboard/mod.rs:584]: Without seeing the user, it's hard to say but it seems like this shouldn't be `mut`

Comment (Drakulix) [reply src/input/keyboard/mod.rs:584]: Sure, it needs to be. It's an `FnMut`.

**Comment by Drakulix** [comment]
Replaced with https://github.com/Smithay/smithay/pull/880

**Pull request closed without merging**